processing/src/main/java/io/druid/segment/serde/DictionaryEncodedColumnPartSerde.java

```diff
@@ -157,67 +335,114 @@ public void write(WritableByteChannel channel) throws IOException
   }
 
   @Override
-  public ColumnPartSerde read(ByteBuffer buffer, ColumnBuilder builder, ColumnConfig columnConfig)
+  public ColumnPartSerde read(
+      ByteBuffer buffer, ColumnBuilder builder, ColumnConfig columnConfig
+  )
   {
-    final boolean isSingleValued = buffer.get() == 0x0;
-    final GenericIndexed<String> dictionary = GenericIndexed.read(buffer, GenericIndexed.stringStrategy);
-    final VSizeIndexedInts singleValuedColumn;
-    final VSizeIndexed multiValuedColumn;
+    final VERSION rVersion = VERSION.fromByte(buffer.get());
+    final int rFlags;
+
+    if(rVersion.compareTo(VERSION.COMPRESSED) >= 0 ) {
+      rFlags = buffer.getInt();
+    } else {
+      rFlags = rVersion.equals(VERSION.UNCOMPRESSED_MULTI_VALUE) ?
+              Feature.MULTI_VALUE.getMask() :
+              NO_FLAGS;
+    }
 
+    final boolean hasMultipleValues = Feature.MULTI_VALUE.isSet(rFlags);
+    if(rVersion.equals(VERSION.COMPRESSED) && hasMultipleValues) {
```

Comment:
Is this combination ever going to actually be serialized?  Or is this just defensive?